spec/controllers/orders_controller_spec.rb

```diff
@@ -31,224 +28,263 @@
     it "assigns @order" do
       get :cart
 
-      expect(assigns(:order)).to eq(@order)
+      expect(assigns(:order)).to eq(test_order)
     end
   end
 
-  describe "POST #add_to_cart" do
+  context "GET #checkout" do
     before :each do
-      @order = Order.create
-      session[:order_id] = @order.id
+      session[:order_id] = test_order.id
+      OrderItem.create(order_id: test_order.id, product_id: test_product.id, quantity_ordered: 5)
+    end
+
+    it "is a success" do
+      get :checkout
+
+      expect(response).to be_success
+      expect(response).to have_http_status(200)
+    end
+
+    it "renders the checkout template" do
+      get :checkout
 
-      @product = Product.create(name: 'a', price: 1, seller_id: 1, stock: 1)
+      expect(response).to render_template("checkout")
+    end
+
+    it "assigns @order" do
+      get :checkout
+
+      expect(assigns(:order)).to eq(test_order)
+    end
+  end
+
+  describe "POST #add_to_cart" do
+    before :each do
+      session[:order_id] = test_order.id
     end
 
     it "assigns @order" do
-      post :add_to_cart, id: 1
+      post :add_to_cart, { id: test_product.id }
 
-      expect(assigns(:order)).to eq(@order)
+      expect(assigns(:order)).to eq(test_order)
     end
 
     it "assigns @product" do
-      post :add_to_cart, id: 1
+      post :add_to_cart, { id: test_product.id }
 
-      expect(assigns(:product)).to eq(@product)
+      expect(assigns(:product)).to eq(test_product)
     end
 
-    it "redirects to the product #show page after save" do
-      # OrderItem.create(product_id: @product.id, order_id: session[:order_id], quantity_ordered: 1)
-      post :add_to_cart, id: 1
+    it "redirects to product#show" do
```

Comment:
I don't think we need to be that explicit. That's the expected behavior for our POST #add_to_cart... I don't know how I'd even explain it for the purposes of testing. To me, that's the same thing as explaining why `render :whatever` is necessary.